system/web/services/RoutingService.cfc

```diff
@@ -315,7 +315,11 @@ component extends="coldbox.system.web.services.BaseService" accessors="true"{
 				if( log.canDebug() ){
 					log.debug( "Invalid HTTP Method detected: #httpMethod#", routeResults.route );
 				}
-			}
+            }
+
+            if ( isStruct( routeResults.route.action ) && structIsEmpty( routeResults.route.action ) ) {
```

Comment:
What is the purpose of this code?

Reply:
The code below only knows how to deal with empty string actions, not empty structs.  The changes to merging routes can introduce an empty struct.  This code resets an empty struct to an empty string so it gets handled by the wildcard or not found routing below.

Reply:
Wouldn't it be best if on line 325 we just add the second check instead of a reset.  I think we should just check if the action is a struct and if it is empty.

Reply:
@elpete